pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java

```diff
@@ -749,6 +758,76 @@ public void testSearch()
     assertNoError(jsonNode);
   }
 
+  @Override
+  protected String getTableName() {
+    return _tableName;
+  }
+
+  @Test
+  public void testWithDatabaseContext()
+      throws Exception {
+    try {
+      _tableName = "db1." + DEFAULT_TABLE_NAME;
+      String defaultCol = "ActualElapsedTime";
+      String customCol = "ActualElapsedTime_2";
+      Schema schema = createSchema();
+      schema.addField(new MetricFieldSpec(customCol, FieldSpec.DataType.INT));
+      addSchema(schema);
+      TableConfig tableConfig = createOfflineTableConfig();
+      assert tableConfig.getIndexingConfig().getNoDictionaryColumns() != null;
+      List<String> noDicCols = new ArrayList<>(DEFAULT_NO_DICTIONARY_COLUMNS);
+      noDicCols.add(customCol);
+      tableConfig.getIndexingConfig().setNoDictionaryColumns(noDicCols);
+      IngestionConfig ingestionConfig = new IngestionConfig();
+      ingestionConfig.setTransformConfigs(List.of(new TransformConfig(customCol, defaultCol)));
+      tableConfig.setIngestionConfig(ingestionConfig);
+      addTableConfig(tableConfig);
+
+      // Create and upload segments to 'db1.mytable'
+      TestUtils.ensureDirectoriesExistAndEmpty(_segmentDir, _tarDir);
+      ClusterIntegrationTestUtils.buildSegmentsFromAvro(_avroFiles, tableConfig, schema, 0, _segmentDir, _tarDir);
+      uploadSegments(getTableName(), _tarDir);
+
+      // Wait for all documents loaded
+      waitForAllDocsLoaded(600_000L);
+
+      // default database check. No database context passed
+      checkQueryResultForDBTest(defaultCol, DEFAULT_TABLE_NAME);
```

Comment:
Could we split these checks into different test methods? Right now if this check fails, we don't test the others, which means it could be more difficult to find the issue.

Reply:
Wanted to get all the checks done with a one time table setup.

Reply:
Couldn't we setup the table once on startup?